### ISSUE TITLE
The first page of a ViewPager doesn't work when setAdapter is asynchronized

### ISSUE THREAD

**Issue by junyuecao** [henrytao-me/smooth-app-bar-layout#59]
The data of my adapter of the ViewPager is fetched from a asynchronized API so setAdapter will be called after the BaseBehavior's onMeasureChild. So that the first page of the ViewPager does not work.
Once I switched between pages of the ViewPager everything works fine.

**Comment by henrytao-me** [comment]
Can you try [syncOffset](https://github.com/henrytao-me/smooth-app-bar-layout/blob/master/smooth-app-bar-layout%2Fsrc%2Fmain%2Fjava%2Fme%2Fhenrytao%2Fsmoothappbarlayout%2FSmoothAppBarLayout.java#L151) or [syncOffsetDelayed](https://github.com/henrytao-me/smooth-app-bar-layout/blob/master/smooth-app-bar-layout%2Fsrc%2Fmain%2Fjava%2Fme%2Fhenrytao%2Fsmoothappbarlayout%2FSmoothAppBarLayout.java#L162) after adding data into list?

**Comment by junyuecao** [comment]
I've tried but it doesn't work.

```
private void onSyncOffset(CoordinatorLayout coordinatorLayout, AppBarLayout child) {
    log("onSyncOffset | %b", vScrollTarget != null);
    if (vScrollTarget != null) {
      onSyncOffset(coordinatorLayout, child, vScrollTarget);
    }
  }
```

The vScrollTarget is null here. I think the root cause is from [here](https://github.com/henrytao-me/smooth-app-bar-layout/blob/master/smooth-app-bar-layout%2Fsrc%2Fmain%2Fjava%2Fme%2Fhenrytao%2Fsmoothappbarlayout%2FBaseBehavior.java#L241) . The `vViewPager.getCurrentItem()` is 0 but the ViewPager does not have pages. 
So `vScrollTarget` is null  at the first time of onViewPagerSelected called:

```
private void onViewPagerSelected(CoordinatorLayout coordinatorLayout, AppBarLayout child, int position) {
    log("onSyncOffset | %b", vViewPager.getAdapter() instanceof PagerAdapter);
    if (vViewPager.getAdapter() instanceof PagerAdapter) {
      PagerAdapter adapter = (PagerAdapter) vViewPager.getAdapter();
      vScrollTarget = getScrollTarget(adapter.getScrollView(position));
      onViewPagerSelected(coordinatorLayout, child, vScrollTarget, vViewPager, position);
    }
  }
```

**Comment by henrytao-me** [comment]
I think this issue is fixed in v1.0.2 and above. Check it out. @junyuecao
